text/0015-status-update.md

```diff
@@ -0,0 +1,397 @@
+`LayoutVerified` with hand-written bounds checks, but for now I prefer the
+ergonomics of this approach.
+
+#### #[attributes]
+
+Here we start to see the first 'attributes': these are various declarations that
+occur in brackets, preceded by a pound sign, like: `#[..]`. In this example the
+entire table has the `#[offset_host]` annotation, and the `lang_sys_records`
+field has the `#[count()]` annotation. The first of these tells the macro that
+this table is used as the base for some offsets (which means we need to hold on
+to our source data, and implement methods to resolve offsets) and the second
+annotation tells the macro that length of the `lang_sys_records` field is
+determined by the value of the `lang_sys_count` field.
+
+There are other attributes available; the exact set will surely continue to
+evolve. Overall, though, this pattern lets us express the majority of the
+patterns I have encountered so far. As a fallback, we can represent certain
+fields as just byte slices, and then hand-write code for accessing them at
+runtime.
+
+#### version enums
+
+One final thing that the macro currently does is generate enums for tables that
+have multiple version/formats. Let's use coverage tables as an example:
+
+```rust
+    CoverageFormat1<'a> {
+        /// Format identifier — format = 1
+        coverage_format: BigEndian<u16>,
+        /// Number of glyphs in the glyph array
+        glyph_count: BigEndian<u16>,
+        /// Array of glyph IDs — in numerical order
+        #[count(glyph_count)]
+        glyph_array: [BigEndian<u16>],
+    }
+
+    CoverageFormat2<'a> {
+        /// Format identifier — format = 2
+        coverage_format: BigEndian<u16>,
+        /// Number of RangeRecords
+        range_count: BigEndian<u16>,
+        /// Array of glyph ranges — ordered by startGlyphID.
+        #[count(range_count)]
+        range_records: [RangeRecord],
+    }
+
+    #[format(u16)]
+    enum CoverageTable<'a> {
+        #[version(1)]
+        Format1(CoverageFormat1<'a>),
+        #[version(2)]
+        Format2(CoverageFormat2<'a>),
+    }
+```
+
+The relevant part is the enum declaration at the bottom. This generates an enum
+that is initialized with a slice of bytes; it reads a single `u16` (specified by
+the `format` attribute) and then based on that value it generates one of the
+variants based on the specified `version` attribute.
+
+#### preliminary codegen
+
+Writing all of these macro declarations is still a lot of work, so there is a
+preliminary step to help out: this step takes as input [text files][] containing
+tables and records copied directly from [the spec][], with a few added
+annotations. This tool generates the vast majority of the code for the table
+definitions, which are then cleaned up manually.
+
+
+## Findings & conclusions
+
+This approach has worked very well for generating types suitable for shaping,
+but is not trivially adaptable to the compilation & modification case.
```

Comment:
Could you elaborate on issues related to use for compilation and/or modification?

Reply:
I think I touch on this below; if it's still not clear let me know and I'll try to fix it up a bit. :)